cpp/daal/src/algorithms/dtrees/gbt/classification/gbt_classification_predict_dense_default_batch_impl.i

```diff
@@ -178,8 +178,103 @@ public:
 protected:
     services::Status predictByAllTrees(size_t nTreesTotal, size_t nClasses, const DimType & dim);
 
-    void predictByTrees(algorithmFPType * res, size_t iFirstTree, size_t nTrees, size_t nClasses, const algorithmFPType * x);
-    void predictByTreesVector(algorithmFPType * val, size_t iFirstTree, size_t nTrees, size_t nClasses, const algorithmFPType * x);
+    template <bool hasUnorderedFeatures, bool hasAnyMissing>
+    using Dispetcher_t = gbt::prediction::internal::PredictDispetcher<hasUnorderedFeatures, hasAnyMissing>;
+    template <bool hasUnorderedFeatures, bool hasAnyMissing>
+    void predictByTrees(algorithmFPType * val, size_t iFirstTree, size_t nTrees, size_t nClasses, const algorithmFPType * x,
+                        const Dispetcher_t<hasUnorderedFeatures, hasAnyMissing>& dispetcher);
+    template <bool hasUnorderedFeatures, bool hasAnyMissing>
+    void predictByTreesVector(algorithmFPType * val, size_t iFirstTree, size_t nTrees, size_t nClasses, const algorithmFPType * x,
+                              const Dispetcher_t<hasUnorderedFeatures, hasAnyMissing>& dispetcher);
+
+    template <bool val>
+    struct BooleanConstant {
+        typedef BooleanConstant<val> type;
+    };
+
+    inline void updateResult(algorithmFPType * res, algorithmFPType * val, size_t iRow, size_t i, size_t nClasses, BooleanConstant<true> dispetcher) {
+        res[iRow + i] = algorithmFPType(getMaxClass(val + i * nClasses, nClasses));
+    }
+
+    inline void updateResult(algorithmFPType * res, algorithmFPType * val, size_t iRow, size_t i, size_t nClasses, BooleanConstant<false> dispetcher) {}
```

Comment:
Same typo (happens in even more places that I won't comment). `dispetcher` -> `dispatcher`.

Why is the implementation empty?

Reply:
In fact we have 4 such comparisons. And they are in the critical section of the code. Such micro optimization become important for test cases with small number of rows. For example, inference speed of letters was increased on 30% for xgboost model builders. Unfortunately for testcases with large number of rows performance gain doesn't exceed few precents.